src/Models/NonhydrostaticModels/update_hydrostatic_pressure.jl

```diff
@@ -24,8 +30,6 @@ function update_hydrostatic_pressure!(pHY′, arch, grid, buoyancy, tracers)
     # Fill halo regions for pressure
     wait(device(arch), pressure_calculation)
 
+    fill_halo_regions!(pHY′, arch)
```

Comment:
@glwagner, is this OK here? It was outside in the `update_state!` but we only want it when z is *not* flat.

Reply:
It's probably best here, rather than in `update_state!`.